plugin/mutate/source/dextool/plugin/mutate/backend/report/html/page_tree_map.d

```diff
@@ -0,0 +1,100 @@
+module dextool.plugin.mutate.backend.report.html.page_tree_map;
+
+import std.json : JSONValue;
+import arsd.dom : Document, Element, require, Table, RawSource;
+
+import dextool.plugin.mutate.backend.report.html : FileIndex;
+import dextool.plugin.mutate.backend.report.html.tmpl : tmplBasicPage;
+
+/* 
+* A JSON-like object that is more easily manipulated than JSONValue.
+* Has operator overloads for "index", "index assign" and "in" allowing the children list to be private
+* and cleaner construction of the object. 
+*/
+class JSONLike {
+    string name;
+    private JSONLike[string] children;
+    long locs = -1;
+    double score = -1;
+
+    JSONValue toJSONValue() {
+        JSONValue root = ["name" : name];
+
+        if (children.length) {
+            JSONValue[] descendants = [];
+            foreach (c; children) {
+                descendants ~= c.toJSONValue();
+            }
+            root.object["children"] = JSONValue(descendants);
+        }
+        if (locs != -1)
+            root.object["locs"] = JSONValue(locs);
+
+        if (score != -1)
+            root.object["score"] = JSONValue(score);
+
+        return root;
+    }
+
+    JSONLike opIndex(string i) pure nothrow {
+        return children[i];
+    }
+
+    JSONLike opIndexAssign(JSONLike value, string i) pure nothrow {
+        children[i] = value;
+        return value;
+    }
+
+    bool opBinaryRight(string op)(string a) {
+        if (op == "in") {
+            if (a in children)
+                return true;
+            return false;
+        }
+        return false;
+    }
+
+}
+
+@trusted auto makeTreeMapJSON(FileIndex[] files) {
```

Comment:
Is this attribute needed? I think everything is actually `@safe` in this file.
Recommended reading: [memory safety in D](https://dlang.org/spec/memory-safe-d.html)